### PR TITLE
[WIP] implement HTML representation

### PR DESCRIPTION
**Pull request by tomkralidis**
Initial minimal scaffolding for review and comment.  Notes:

- idea is to pass wfs3 types into HTML template
- HTML templates are strings (not files) for ease of deployment
- `f=text/html` overrides HTTP Accept header

### CODE REVIEW

**Comment by JivanAmara** [comment]
Check out these changes & consider if we should pass config to MarshalHTML() or just the pieces we want:
https://gist.github.com/JivanAmara/f06f151c20cd8fec0e02dba027d8052d#file-conformance-go

**Comment by tomkralidis** [comment]
Thansk @JivanAmara. Comments:

- will update to have struct methods
- same for `map[string]string`
- what's the downside of passing the config?  What is there now is minimal, but things like service metadata are valuable to the HTML representation.  What about passing sections of the config as an option?
- I would stay away from things like https://gist.github.com/JivanAmara/f06f151c20cd8fec0e02dba027d8052d#file-conformance-go-L30 given we want the template to render HTML, not code per se.

**Comment by JivanAmara** [comment]
The down sides to passing the config is an unnecessary coupling of the config w/ the templates with pretty clear opportunity for future developers to trip on it.  I think that the stuff we want to pass from the config probably belongs to a "page" template that the top-level element templates extend.

For the wfs3 types, yes since each has a dedicated template the first context variable should be the time, for anything needed outside of that, or for the html of properties that are also wfs3 types, a map[string]string context type.

The example at https://gist.github.com/JivanAmara/f06f151c20cd8fec0e02dba027d8052d#file-conformance-go-L30 isn't how I think this template should be rendered, but is to illustrate how this will generalize for contained types.  For example, a number of wfs3 types use the wfs3 Link in their properties, such as CollectionInfo.  With this approach we don't re-create how to render a Link in the CollectionInfo template, but call Link.MarshalHTML() and pass the result to the CollectionInfo template to put where it would like it.  It makes the rendering of the elements as composable as the elements themselves & avoids repeating logic for rendering some types in multiple places.

File: util/util.go [hunk 1]
```diff
@@ -74,3 +76,16 @@ func DefaultGpkg() string {
...
+func RenderTemplate(templateString string, data map[string]interface{}) ([]byte, error) {
+	var tpl bytes.Buffer
+	t := template.New("template")
+	t, _ = t.Parse(templateString)
+
+	if err := t.Execute(&tpl, data); err != nil {
+		return tpl.Bytes(), err
+	}
+
+	// FIXME: should be a better way
```
Comment (JivanAmara) [inline util/util.go]: What's your concern here?

**Comment by tomkralidis** [comment]
Closed in lieu of #97

**Pull request closed without merging**